### PR TITLE
Disc 172/different header on full record

### PR DESCRIPTION
**Pull request by jesperlauridsen**
This should be a decent first stab at the full record page. I think there's only minor stuff missing, and I'll do JIRAs on that as we run along.

### CODE REVIEW

File: src/components/search/result-component.ts [hunk 1]
```diff
@@ -71,7 +71,9 @@ class ResultComponent extends HTMLElement {
 		duration && (duration.textContent = 'VARIGHED');
 
 		const thumb = this.shadow.querySelector('.image-item') as HTMLImageElement;
-		thumb && (thumb.src = resultData.thumbnail);
+		if (resultData.thumbnail !== undefined) {
+			thumb && (thumb.src = resultData.thumbnail);
+		}
```
Comment (jorntx) [inline src/components/search/result-component.ts:76]: This takes care of DISC-199 - nice!

**Comment by jorntx** [comment]
I like the overall direction this is heading. Very easy to abstract global web components later on. The layout is very well aligned with the design proposal.

File: src/components/records/BroadcastRecord.vue [hunk 2]
```diff
@@ -1,24 +1,62 @@
...
-	<div class="">
+	<div class="broadcast-record">
 		<VideoPlayer></VideoPlayer>
-		<div class="title-box">
-			<h2>{{ recordData.name[0].value }}</h2>
+		<div class="boardcast-record-data">
+			<div class="main-record-data">
+				<div class="record-data">
+					<h2>{{ recordData.name[0].value }}</h2>
+					<p>{{ recordData.description }} ALSO, LOREM IPSUM PLEASE</p>
```
Comment (jorntx) [inline src/components/records/BroadcastRecord.vue:8]: I guess the descriptions in data are very short

File: src/components/records/BroadcastRecord.vue [hunk 3]
```diff
@@ -1,24 +1,62 @@
...
-					{{ recordData.description }}
-				</p>
+		<div class="extra-record-data">
+			<div class="accordion">
+				<kb-accordion
+					first="true"
+					title="Yderligere information om videoen"
+					expanded="Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi sit amet quam felis. Curabitur dui augue, auctor eu sodales sit amet, porta vel dolor. Ut libero purus, malesuada ut tincidunt non, auctor nec magna. Quisque interdum, libero vitae varius tempor, ipsum sapien tempor tellus, et aliquet ex enim in neque. Donec lacinia justo urna, et imperdiet tellus sodales sit amet. Nulla nec aliquam nunc. Nam pretium suscipit posuere. Pellentesque tincidunt auctor mattis.
+					
+Mauris non ligula a urna dapibus egestas eget at sem. Sed ac nulla ex. Cras quis ligula at nulla tincidunt consequat. Aliquam arcu est, malesuada non sapien at, malesuada tempus nulla. Etiam faucibus condimentum leo, eget euismod eros cursus fermentum. Fusce eget arcu non nulla vulputate aliquet eget id velit. Integer ipsum tellus, tempus quis elementum id, dictum vitae libero. Nullam at convallis lectus. Morbi pellentesque eget nisi id tempor."
```
Comment (jorntx) [inline src/components/records/BroadcastRecord.vue:47]: Looking forward to seeing the real data - nobody has actually promised us such beefy descriptions

File: src/components/records/BroadcastRecord.vue [hunk 4]
```diff
@@ -1,24 +1,62 @@
...
-					<span class="broadcast-duration">({{ getBroadcastDuration(recordData.duration) }})</span>
+			<div class="related-content">
+				<h3>Relateret indhold</h3>
+				<div class="related-records">
+					<div class="related-record"><kb-spotcomponent></kb-spotcomponent></div>
+					<div class="related-record"><kb-spotcomponent></kb-spotcomponent></div>
+					<div class="related-record"><kb-spotcomponent></kb-spotcomponent></div>
+					<div class="related-record"><kb-spotcomponent></kb-spotcomponent></div>
+					<div class="related-record"><kb-spotcomponent></kb-spotcomponent></div>
+					<div class="related-record"><kb-spotcomponent></kb-spotcomponent></div>
```
Comment (jorntx) [inline src/components/records/BroadcastRecord.vue:58]: Very nice with the reusable spot component here

File: src/components/records/BroadcastRecord.vue [hunk 5]
```diff
@@ -40,7 +80,14 @@ export default defineComponent({
...
 		},
 	},
+	created() {
+		console.log('object', this.recordData);
+	},
 	methods: {
+		getCurrentUrl() {
+			//make function to copy to clipboard.
+			console.log('YHEARRAP');
+		},
```
Comment (jorntx) [inline src/components/records/BroadcastRecord.vue:90]: Made a jira for this - DISC-222

File: src/components/spot-component.ts [hunk 6]
```diff
@@ -0,0 +1,133 @@
+class SpotComponent extends HTMLElement {
+	shadow: ShadowRoot;
+	data: unknown | undefined;
```
Comment (jorntx) [inline src/components/spot-component.ts:3]: Let's remember to get some typing in here when we now the real data

File: src/components/viewers/AudioVideo/VideoPlayer.vue
Comment (jorntx) [inline src/components/viewers/AudioVideo/VideoPlayer.vue:1]: Feel some merge hurt coming my way!!!!!

**Review by jorntx — COMMENTED** [review]
Looks very good. I have only minor comments.

**Review by jorntx — APPROVED** [review]
Since my last look at it this there were changes in BroadcastRecord and spot-component and it all looks fine and I took it for a spin - all good.

I have only minor comments - nothing that blocks a merge.

I think what is left now is more nitty gritty work when we get to a place where we can make definitive design decisions and then we can polish this.

I can see that you put a lot of work into this and I think it turned out very well @jesperlauridsen! This is a solid proposal on how things should look and work also in regards to the responsive part. I tip my hat.

File: src/components/spot-component.ts [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+		if (this.title) {
+			const titleHolder = this.shadow.querySelector('.record-title');
+			titleHolder && (titleHolder.innerHTML = this.title);
+			const symbol = this.shadow.querySelector('.type-symbol');
+			symbol && (symbol.innerHTML = 'play_arrow');
+		}
+		if (this.data) {
+			//put in the data
+		}
+	}
```
[the remark this answers is not in the document]
Comment (jesperlauridsen) [reply src/components/spot-component.ts:25]: Very much so!

**Pull request merged**